### PR TITLE
Implemented search on new comet endpoint

### PR DESCRIPTION
**Pull request by dmcollom**
The MPC added, on request, an endpoint to enable searching for comets in addition to the existing asteroid search endpoint. In this PR:

* MPC module can query both asteroid endpoint and comet endpoint, and now requires a target_type parameter
* Refactor to reduce code reuse

### CODE REVIEW

**Comment by keflavich** [comment]
Is this ready for final review?

**Comment by dmcollom** [comment]
Yep, it's been ready!

File: astroquery/mpc/core.py [hunk 1]
```diff
@@ -9,39 +9,46 @@
 
 @async_to_sync
 class MPCClass(BaseQuery):
-    MPC_URL = 'http://' + conf.server + '/web_service/search_orbits'
-    # The authentication credentials for the MPC web service are publicly available and
-    # can be openly viewed on the documentation page at https://minorplanetcenter.net/web_service/
+    MPC_URL = 'http://' + conf.server + '/web_service'
```
Comment (keflavich) [inline astroquery/mpc/core.py]: just to check: is it correct that this is http, not https?

**Comment by keflavich** [comment]
OK, sorry about the delay.  It helps to leave a message when (you think) you're done editing; we have been getting a little overwhelmed by the large number of contributions recently.

**Pull request merged**